todo_app/app.py

```diff
@@ -1,30 +1,190 @@
+    return redirect(url_for('moved'))
+
+@app.route('/complete/<card>')
+def complete_item(card):
+    myboards = myTrello()
+    myboards.get_trello_boards_name_and_ref()
+    myboards.get_trello_lists_on_board('DevOps Module 2')
+    lists = myboards.get_my_list_info('DevOps Module 2')
+    list_keys = list(lists.keys())
+    doing_list = list_keys[1]
+    myboards.get_trello_cards_on_list(doing_list)
+    doing_cards = myboards.get_cards_on_doing_list(doing_list)
+    id = doing_cards[card]
+    body = {'idList': '5f69d2475add26219791fc05'}
+    requests.put(base_url + 'cards/' + id + '?', params=payload, data=body)
+    return redirect(url_for('moved'))
+
+@app.route('/redo_doing/<card>')
+def redo_doing_item(card):
+    myboards = myTrello()
+    myboards.get_trello_boards_name_and_ref()
+    myboards.get_trello_lists_on_board('DevOps Module 2')
+    lists = myboards.get_my_list_info('DevOps Module 2')
+    list_keys = list(lists.keys())
+    doing_list = list_keys[1]
+    myboards.get_trello_cards_on_list(doing_list)
+    doing_cards = myboards.get_cards_on_doing_list(doing_list)
+    id = doing_cards[card]
+    body = {'idList': '5f69d2471e787e8f0dd62f93'}
+    requests.put(base_url + 'cards/' + id + '?', params=payload, data=body)
+    return redirect(url_for('moved'))
+
+@app.route('/restart/<card>')
+def restart_item(card):
+    myboards = myTrello()
+    myboards.get_trello_boards_name_and_ref()
+    myboards.get_trello_lists_on_board('DevOps Module 2')
+    lists = myboards.get_my_list_info('DevOps Module 2')
+    list_keys = list(lists.keys())
+    done_list = list_keys[2]
+    myboards.get_trello_cards_on_list(done_list)
+    done_cards = myboards.get_cards_on_done_list(done_list)
+    id = done_cards[card]
+    body = {'idList': '5f69d2471e787e8f0dd62f93'}
+    requests.put(base_url + 'cards/' + id + '?', params=payload, data=body)
+    return redirect(url_for('moved'))
+
+@app.route('/redo/<card>')
+def redo_item(card):
+    myboards = myTrello()
+    myboards.get_trello_boards_name_and_ref()
+    myboards.get_trello_lists_on_board('DevOps Module 2')
+    lists = myboards.get_my_list_info('DevOps Module 2')
+    list_keys = list(lists.keys())
+    done_list = list_keys[2]
+    myboards.get_trello_cards_on_list(done_list)
+    done_cards = myboards.get_cards_on_done_list(done_list)
+    id = done_cards[card]
+    body = {'idList': '5f69d24735e43368d5ec3ac0'}
+    requests.put(base_url + 'cards/' + id + '?', params=payload, data=body)
+    return redirect(url_for('moved'))
+
+@app.route('/moved', methods=['GET'])
+def moved():
+    return render_template('move_confirmed.html')   
+
+@app.route('/lists/<board_name>')
+def go_to_board_lists(board_name):
+    myboards = myTrello()
+    myboards.get_trello_boards_name_and_ref()
+    myboards.get_trello_lists_on_board(board_name)
+    lists = myboards.get_my_list_info(board_name)
+    return render_template('my_lists.html', lists=lists)
```

Comment:
I've taken the liberty of a quick attempt at rewriting this file in a neater way. Let me know if you'd like to discuss any aspect of it. It's not perfect but 🤷 

Also, the functions I put at the top of the file could be wrapped in a class but I haven't bothered as I've already spent slightly longer than I ought to spend on a review...

index.html and boardtasks.html obviously need rewriting to match my changes. See those files for comments with code
```
base_url = 'https://api.trello.com/1/'
def get_auth_params():
    f = open("todo_app/Trello_API_Keys.txt", "r").read().split("\n")
    return {'key': f[0], 'token': f[1]}

def get_boards():
    return requests.get(base_url + 'members/me/boards?', params=get_auth_params()).json()

def get_trello_lists_on_board(board_name):
    board = next(board for board in get_boards() if board['name'] == board_name)
    lists = requests.get(base_url +'boards/' + board['id'] + '/lists?', params=get_auth_params()).json()
    for list in lists:
        list['cards'] = get_trello_cards_on_list(list['id'])
    return lists

def get_trello_cards_on_list(list_id):
    return requests.get(base_url +'lists/' + list_id + '/cards?', params=get_auth_params()).json()

@app.route('/', methods=['GET'])
def index():
    return render_template('Index.html', boards=get_boards())

@app.route('/board/<board_name>', methods=['GET'])
def go_to_board_tasks(board_name):
    lists = get_trello_lists_on_board(board_name)
    return render_template('boardtasks.html', board_name=board_name, lists=lists)

@app.route('/board/<board_name>/new-task', methods=['POST'])
def create_board_task(board_name):
    payload_data = get_auth_params()
    payload_data['idList'] = request.form['idList']
    payload_data['name'] = request.form['title']
    requests.post(base_url + 'cards', params=payload_data)
    return redirect(url_for('go_to_board_tasks', board_name=board_name))

@app.route('/move-card/<card_id>/to/<list_id>')
def move_card(card_id, list_id):
    params = get_auth_params()
    params['idList'] = list_id
    requests.put(base_url + 'cards/' + card_id + '?', params=params)
    return redirect(url_for('moved'))

@app.route('/moved', methods=['GET'])
def moved():
    return render_template('move_confirmed.html')   
```